daemons/controld/controld_based.c

```diff
@@ -46,6 +46,28 @@ do_cib_replaced(const char *event, xmlNode * msg)
     register_fsa_input(C_FSA_INTERNAL, I_ELECTION, NULL);
 }
 
+void
+controld_disconnect_cib_manager(void)
+{
+    CRM_ASSERT(fsa_cib_conn != NULL);
+
+    crm_info("Disconnecting from the CIB manager");
+
+    controld_clear_fsa_input_flags(R_CIB_CONNECTED);
+
+    fsa_cib_conn->cmds->del_notify_callback(fsa_cib_conn, T_CIB_REPLACE_NOTIFY, do_cib_replaced);
+    fsa_cib_conn->cmds->del_notify_callback(fsa_cib_conn, T_CIB_DIFF_NOTIFY, do_cib_updated);
+    cib_free_callbacks(fsa_cib_conn);
+    if (fsa_cib_conn->state != cib_disconnected) {
+        /* Does not require a set_slave() reply to sign out from based. */
+        fsa_cib_conn->cmds->set_slave(fsa_cib_conn, cib_scope_local | cib_discard_reply);
+        fsa_cib_conn->cmds->signoff(fsa_cib_conn);
+    }
+    verify_stopped(fsa_state, LOG_WARNING);
```

Comment:
verify_stopped() should stay in crmd_exit() only. It's a good example of why names should be more descriptive ... its intent is to verify that all resources are stopped, before disconnecting from the executor.